### PR TITLE
Adds lint and test pre-commit hooks

### PR DESCRIPTION
**Pull request by mcpolandc**
It occurred to me while reading the contribution guide that the project might benefit from automatic push hooks to lint and test.

### CODE REVIEW

**Comment by JedWatson** [comment]
@mcpolandc thanks! good idea.

Thoughts on using `husky` and `lint-staged` with a `pre-commit` hook though? (instead of pre-push, errors would be caught before commit)

**Comment by mcpolandc** [comment]
Hi @JedWatson! Yeah, running the linter on only the staged files when committing makes a lot of sense. I will make changes to this effect 😄 

Just extending on this... I wonder if we should maybe add the test cover check at the commit stage as well. Maybe `husky`, `lint-staged` or some other package can provide this functionality (exit with 1 if test coverage has dropped, thus failing commit). What are your thoughts on this?

**Comment by mcpolandc** [comment]
Changed this to run lint (only staged js files) and test scripts on commit rather than push.

File: package.json [hunk 1]
```diff
@@ -97,5 +100,8 @@
     "react-component",
     "select",
     "ui"
-  ]
+  ],
+  "lint-staged": {
+    "*.js": ["eslint", "git add"]
```
Comment (JedWatson) [inline package.json]: What is the point of `git add` here? unless `eslint` is being run with the `--fix` flag (which it's not and I'm happy to leave it that way) this wouldn't be necessary; and it has the side-effect of preventing you from committing only staged chunks of a file without making it an obvious thing...

Surely just `eslint` on its own would be sufficient (and safer!) here?

**Comment by JedWatson** [comment]
Thanks @mcpolandc! Left a note there for feedback but otherwise this looks great!

File: package.json (hunk 1 above)
Comment (mcpolandc) [reply package.json]: I knew that I didn't want to have it fix linter errors automatically but I didn't really understand what git add was doing. Reading further down the readme of lint-staged helped with that 😛. I totally agree with what you're saying here so making the change.

**Comment by JedWatson** [comment]
Great, thanks!

**Pull request merged**